src/test/java/emissary/core/BaseDataObjectTest.java

```diff
@@ -1319,4 +1322,137 @@ void testExtractedRecordClone() {
             fail("Clone method should have been called", ex);
         }
     }
+
+    @Test
+    void testChannelFactoryInArrayOutNoSet() throws IOException {
```

Comment:
I'm not sure the names of these 3 new tests provide any insight into what they're trying to validate.  If there's too much to convey via the method name(s), we should add some clarifying comments, either inline or Javadoc